core/src/main/java/com/datastax/oss/driver/api/core/data/CqlVector.java

```diff
@@ -15,25 +15,31 @@
  */
 package com.datastax.oss.driver.api.core.data;
 
-import com.datastax.oss.driver.shaded.guava.common.base.Joiner;
-import com.datastax.oss.driver.shaded.guava.common.collect.ImmutableList;
-import com.datastax.oss.driver.shaded.guava.common.collect.Iterators;
-import java.util.Arrays;
+import java.util.*;
 
-/** An n-dimensional vector defined in CQL */
-public class CqlVector<T> {
+/**
+ * An n-dimensional vector defined in CQL. You can use either {@link CqlVector#builder()} for an
+ * iterable interface or {@link #CqlVector(List)} directly. Once created, a CqlVector is immutable.
+ */
+public class CqlVector<T> implements Iterable<T> {
 
-  private final ImmutableList<T> values;
+  private final List<T> values;
```

Comment:
Mentioned in my [original review](https://datastax.slack.com/archives/C053MUQHLAK/p1685736386479569?thread_ts=1685666615.431329&cid=C053MUQHLAK) of what became this PR: I don't see any reason for this change.  We're giving up informing the type system that the list in question is immutable for no obvious benefit.

Reply:
Sorry, I wasn't looking in chat for review content after our first convo. I thought it would be here. This is a design choice. Here are the reasons for my suggestion:

I've preferred to avoid dependencies on Guava due to the instability, unneccesary complexity, library bloat, and packaging problems (not strictly backwards compatible, etc) that have arisin in other projects that depend on it. In this case, informing the type system doesn't really do anything, since the users doesn't see this in the CqlVector contract, and the effective behavior is the same. (The implemented type is effectively immutable.) Depending on this library also puts deeper roots into the Guava type system which is starting to overlap significantly with more modern Java idioms. Based on having to disentagle Guava from other projects for the reasons above, I've taken a stricter approach for if and when I would rely on it.

But this isn't something that is important enough to block a PR for, so I'll just follow your lead on the actual change.

Reply:
Are we already using Guava in the driver today?

Reply:
I'll only mention that my comment about having the immutability of that collection reflected in the type system had to do with code internal to CqlVector rather than consumers of that library.  Which leads beautifully into my answer to the question posed by @jbellis :)

The Java driver does use Guava fairly extensively.  In the 3.x driver there are a few spots that returned Guava types directly from methods but this practice has been discontinued for the 4.x driver.  In 4.x we use a shaded version of Guava (an older one actually) and we only use it internally.

Reply:
Understood, but the desire to disentangle Guava as a form of incremental simplification still stands for me, not to protect users directly, but more to make the code easier to maintain over time. This item is mostly an academic discussion on this point, but I wanted to be transparent on my reasons for suggesting the change.